controllers/cloud.redhat.com/clowdjobinvocation_controller.go

```diff
@@ -345,23 +508,43 @@ func (r *ClowdJobInvocationReconciler) cjiToEnqueueUponJobUpdate(a handler.MapOb
 // setCompletedStatus will determine if a CJI has completed all needed Jobs
 func (r *ClowdJobInvocationReconciler) setCompletedStatus(ctx context.Context, cji *crd.ClowdJobInvocation) error {
 
+	jobs := batchv1.JobList{}
+	if err := r.Client.List(ctx, &jobs, client.InNamespace(cji.ObjectMeta.Namespace)); err != nil {
+		return err
+	}
+
+	cji.Status.Completed = getInvocationStatus(&jobs, cji)
+
+	return nil
+}
+
+func getInvocationStatus(jobs *batchv1.JobList, cji *crd.ClowdJobInvocation) bool {
```

Comment:
I would think that a Status, would return something other than a bool, how do we know what true/false means? If you intend to keep it as a bool, you could encode the meaning into the function name. `isInvocationComplete` ?

Reply:
Maybe this is a misnomer then. "Status" is a boolean about the completion of a CJI. It doesn't go into more detail about what worked/what didn't work. I can update to reflect the boolean(ness)? :smile:

Reply:
I think an isDone kinda deal is nice here then